### DISCUSSION TITLE
Known compiler warnings

### DISCUSSION THREAD

**Discussion by mcuee** [avrdudes/avrdude#1811] [General]
Use this to document some known compiler warnings which will not be fixed as of now.

**Comment by mcuee** [comment]
1. macOS

```
  LEX      lexer.c
/Library/Developer/CommandLineTools/usr/bin/make  all-recursive
Making all in .
  CC       avrdude-main.o
  CC       avrdude-whereami.o
  CC       avrdude-developer_opts.o
  CC       libavrdude_a-config_gram.o
  CC       libavrdude_a-lexer.o
lexer.c:2434:38: warning: comparison of integers of different signs: 'unsigned long' and 'int' [-Wsign-compare]
        if (((yy_n_chars) + number_to_move) > YY_CURRENT_BUFFER_LVALUE->yy_buf_size) {
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```
https://github.com/avrdudes/avrdude/pull/1800#issuecomment-2081648027
From @stefanrueger 

>  That's a known issue and we cannot do anything about it, as it's MacOS's flex that generates the code.

**Reply by mcuee** [reply]
This no longer exists as of May 2025 -- looks like github updated the macOS building infrastructure.

**Reply by mcuee** [reply]
On my Mac Mini M1 running latest macOS 15.5 and latest Homebrew, the warning is still there. But anyway, we will live with this.

**Comment by mcuee** [comment]
Windows MSYS2 clang64 build:
```
D:/a/avrdude/avrdude/src/ser_win32.c:436:3: warning: comparison of integers of different signs: 'SOCKET' (aka 'unsigned long long') and 'const int' [-Wsign-compare]
  436 |                 FD_SET(fd->ifd, &rfds);
      |                 ^~~~~~~~~~~~~~~~~~~~~~
D:/a/_temp/msys64/clang64/include/psdk_inc/_fd_types.h:77:40: note: expanded from macro 'FD_SET'
   77 |                 if (((fd_set *)(set))->fd_array[__i] == (fd)) {         \
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^   ~~
D:/a/avrdude/avrdude/src/ser_win32.c:[55](https://github.com/avrdudes/avrdude/actions/runs/8935908735/job/24545265227#step:5:56)7:3: warning: comparison of integers of different signs: 'SOCKET' (aka 'unsigned long long') and 'const int' [-Wsign-compare]
  557 |                 FD_SET(fd->ifd, &rfds);
      |                 ^~~~~~~~~~~~~~~~~~~~~~
D:/a/_temp/msys64/clang64/include/psdk_inc/_fd_types.h:77:40: note: expanded from macro 'FD_SET'
   77 |                 if (((fd_set *)(set))->fd_array[__i] == (fd)) {         \
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^   ~~
2 warnings generated.
```

From @stefanrueger 
https://github.com/avrdudes/avrdude/issues/1541#issuecomment-1783937965

> According to `man FD_SET` the function is called as it should be. I suspect Windows has a different internal representation and therefore triggers the warning. I suspect using `FD_SET((unsigned long long) fd->ifd, &rfds);` might silence the warning and still work, but it could be that different compilers have different macros and that you get warnings in other compilers. As far as I am concerned we could leave this as is.

**Reply by mcuee** [reply]
Take note MSYS2 mingw64 build and ucrt64 build do not generate such warnings. So it could be MSYS2 clang64 specific issue.

**Reply by mcuee** [reply]
* https://github.com/avrdudes/avrdude/issues/1988#issuecomment-2884620157

Looks like nothing we can do about this. So we will have to live with these MSYS2 clang64 warnings.

**Reply by mcuee** [reply]
Yes, we have to live with this one as it is still there with the latest git action build.
https://github.com/avrdudes/avrdude/actions/runs/33599728349/job/100150531155

**Comment by mcuee** [comment]
Windows VS2019 build warnings -- not existing in VS2022 build.

```
  avrpart.c
C:\work\avr\avrdude_test\avrdude_main\src\avrpart.c(618,13): warning C4090: 'function': different 'const' qualifiers [C
:\work\avr\avrdude_test\avrdude_main\build_msvc\src\libavrdude.vcxproj]
  term.c
C:\work\avr\avrdude_test\avrdude_main\src\term.c(1830,3): warning C4090: 'function': different 'const' qualifiers [C:\w
ork\avr\avrdude_test\avrdude_main\build_msvc\src\libavrdude.vcxproj]
C:\work\avr\avrdude_test\avrdude_main\src\term.c(1835,3): warning C4090: 'function': different 'const' qualifiers [C:\w
ork\avr\avrdude_test\avrdude_main\build_msvc\src\libavrdude.vcxproj]
C:\work\avr\avrdude_test\avrdude_main\src\term.c(2423,7): warning C4090: 'function': different 'const' qualifiers [C:\w
ork\avr\avrdude_test\avrdude_main\build_msvc\src\libavrdude.vcxproj]
C:\work\avr\avrdude_test\avrdude_main\src\term.c(2437,5): warning C4090: 'function': different 'const' qualifiers [C:\w
ork\avr\avrdude_test\avrdude_main\build_msvc\src\libavrdude.vcxproj]
  update.c
C:\work\avr\avrdude_test\avrdude_main\src\update.c(273,24): warning C4090: 'function': different 'const' qualifiers [C:
\work\avr\avrdude_test\avrdude_main\build_msvc\src\libavrdude.vcxproj]
C:\work\avr\avrdude_test\avrdude_main\src\update.c(343,28): warning C4090: 'function': different 'const' qualifiers [C:
\work\avr\avrdude_test\avrdude_main\build_msvc\src\libavrdude.vcxproj]
```

**Reply by mcuee** [reply]
@stefanrueger 

Just want to make sure that the warnings generated by VS2019 is harmless and no need to fix. Thanks.

**Reply by stefanrueger** [reply]
@mcuee Thanks for collecting the compiler warnings from various sources!

TLDR; In this instance the warning is given in error.

The long of it: The compiler warns about freeing a pointer that points to pointers to read-only data, something innocuous as
```
#include <stdlib.h>

int main() {
  const char **p = malloc(42 * sizeof *p);
  free(p);
}
```
It is *wrong* to issue a warning here, as `p` *does* point to modifiable memory as per free()'s signature. It is just that the modifiable space is meant to house pointers to r/o data. It might be that this compiler misread `const char **p` as  `char * const *p`?

**Reply by mcuee** [reply]
No more such issues with latest VS2026 build.
https://github.com/avrdudes/avrdude/actions/runs/33599728349/job/100150531169

**Comment by mcuee** [comment]
Latest issue reports and discussions for the upcoming 8.1 release.
* https://github.com/avrdudes/avrdude/issues/1988

Some fixes are included in PR #1989 
* https://github.com/avrdudes/avrdude/issues/1989

Remaining warnings will be deemed acceptable for now.

We will also ignore the warnings for the Python binding.

**Comment by mcuee** [comment]
@stefanrueger 

One more compiler warning for you to take a look at -- with Raspberry Pi 400 running latest version of Raspberry Pi OS 64 bit version.

```
[ 59%] Building C object src/CMakeFiles/swig_avrdude.dir/config.c.o
/home/mcuee/build/avrdude/src/config.c: In function ‘cfg_escapen’:
/home/mcuee/build/avrdude/src/config.c:876:43: warning: comparison is always true due to limited range of data type [-Wtype-limits]
  876 |       if(n > 1 && ((s[1] & 0x80) || (s[1] >= 0 && s[1] <= '9')))
      |                                           ^~

mcuee@raspberrypi400arm64:~ $ uname -a
Linux raspberrypi400arm64 6.12.25+rpt-rpi-v8 #1 SMP PREEMPT Debian 1:6.12.25-1+rpt1 (2025-04-30) aarch64 GNU/Linux

mcuee@raspberrypi400arm64:~ $ lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 12 (bookworm)
Release:	12
Codename:	bookworm

mcuee@raspberrypi400arm64:~ $ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/aarch64-linux-gnu/12/lto-wrapper
Target: aarch64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 12.2.0-14+deb12u1' --with-bugurl=file:///usr/share/doc/gcc-12/README.Bugs --enable-languages=c,ada,c++,go,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-12 --program-prefix=aarch64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-libquadmath --disable-libquadmath-support --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --enable-fix-cortex-a53-843419 --disable-werror --enable-checking=release --build=aarch64-linux-gnu --host=aarch64-linux-gnu --target=aarch64-linux-gnu
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 12.2.0 (Debian 12.2.0-14+deb12u1)
```

**Comment by stefanrueger** [comment]
> One more compiler warning for you to take a look at

Thanks @mcuee. Fixed in git main.

**Comment by mcuee** [comment]
@stefanrueger 

There is still one warning for NetBSD 10.1 release with avrdude 8.1RC. 
Please check if it is worth fixing or not. Thanks.
```
[ 96%] Building C object src/CMakeFiles/avrdude.dir/main.c.o
In file included from /usr/include/ctype.h:100,
                 from /home/mcuee/build/avr/avrdude_81_rc/src/main.c:42:
/home/mcuee/build/avr/avrdude_81_rc/src/main.c: In function 'main':
/home/mcuee/build/avr/avrdude_81_rc/src/main.c:1019:15: warning: array subscript has type 'char' [-Wchar-subscripts]
 1019 |       toupper(*progname), *progname? progname+1: "", AVRDUDE_FULL_VERSION
      |               ^

[mcuee@netbsd10n4200 avrdude_bin]$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/lto-wrapper
Target: x86_64--netbsd
Configured with: /usr/src/tools/gcc/../../external/gpl3/gcc/dist/configure --target=x86_64--netbsd --enable-long-long --enable-threads --with-bugurl=http://www.NetBSD.org/support/send-pr.html --with-pkgversion='NetBSD nb2 20230710' --with-system-zlib --without-isl --enable-__cxa_atexit --enable-libstdcxx-time=rt --enable-libstdcxx-threads --with-diagnostics-color=auto-if-env --with-tune=nocona --with-default-libstdcxx-abi=new --with-mpc-lib=/var/obj/mknative/amd64-x86_64/usr/src/external/lgpl3/mpc/lib/libmpc --with-mpfr-lib=/var/obj/mknative/amd64-x86_64/usr/src/external/lgpl3/mpfr/lib/libmpfr --with-gmp-lib=/var/obj/mknative/amd64-x86_64/usr/src/external/lgpl3/gmp/lib/libgmp --with-mpc-include=/usr/src/external/lgpl3/mpc/dist/src --with-mpfr-include=/usr/src/external/lgpl3/mpfr/dist/src --with-gmp-include=/usr/src/external/lgpl3/gmp/lib/libgmp/arch/x86_64 --enable-tls --disable-multilib --disable-libstdcxx-pch --build=x86_64--netbsd --host=x86_64--netbsd --with-sysroot=/var/obj/mknative/amd64-x86_64/usr/src/destdir.amd64
Thread model: posix
Supported LTO compression algorithms: zlib
gcc version 10.5.0 (nb3 20231008) 

[mcuee@netbsd10n4200 avrdude_bin]$ uname -a
NetBSD netbsd10n4200.localdomain 10.1 NetBSD 10.1 (GENERIC) #0: Mon Dec 16 13:08:11 UTC 2024  mkrepro@mkrepro.NetBSD.org:/usr/src/sys/arch/amd64/compile/GENERIC amd64
```

**Comment by mcuee** [comment]
Known compiler warnings for OpenBSD. We will not fix these warning as per previous discussions.

Tested with OpenBSD 7.7 release and avrdude 8.1RC.
```
[mcuee@openbsdn4200 tools]$ uname -a
OpenBSD openbsdn4200.lan 7.7 GENERIC.MP#0 amd64

[mcuee@openbsdn4200 tools]$ cc -v
OpenBSD clang version 16.0.6
Target: amd64-unknown-openbsd7.7
Thread model: posix
InstalledDir: /usr/bin

[mcuee@openbsdn4200 avrdude_81_rc]$ ./build.sh
...
...
[ 50%] Linking C executable avrdude
ld: warning: main.c:1067 (/home/mcuee/build/avr/avrdude_81_rc/src/main.c:1067)(CMakeFiles/avrdude.dir/main.c.o:(main)): warning: strcpy() is almost always misused, please use strlcpy()
ld: warning: developer_opts.c:504 (/home/mcuee/build/avr/avrdude_81_rc/src/developer_opts.c:504)(CMakeFiles/avrdude.dir/developer_opts.c.o:(dev_output_part_defs)): warning: sprintf() is often misused, please use snprintf()
ld: warning: main.c:676 (/home/mcuee/build/avr/avrdude_81_rc/src/main.c:676)(CMakeFiles/avrdude.dir/main.c.o:(concatpath)): warning: strcat() is almost always misused, please use strlcat()
ld: warning: dryrun.c:559 (/home/mcuee/build/avr/avrdude_81_rc/src/dryrun.c:559)(dryrun.c.o:(dryrun_enable) in archive libavrdude.a): warning: rand() may return deterministic values, is that what you want?
...
```

**Comment by stefanrueger** [comment]
>  warning for NetBSD 10.1 release with avrdude 8.1RC

Fixed (hopefully)  with last commit

**Reply by mcuee** [reply]
> > warning for NetBSD 10.1 release with avrdude 8.1RC
> 
> Fixed (hopefully) with last commit

Yes, no more compiler warnings under NetBSD 10.1.

**Comment by mcuee** [comment]
There will also be warnings for the Python bindings. We will not deal with them as it is related to swig and outside of avrdude control.

Example build log under MSYS2 mingw64.
```
MINGW64 /c/work/avr/avrdude_main
$ ./build.sh
-- Building for: Ninja
-- The C compiler identification is GNU 15.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/msys64/mingw64/bin/cc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found Git: C:/msys64/usr/bin/git.exe (found version "2.50.0")
-- Found FLEX: C:/msys64/usr/bin/flex.exe (found version "2.6.4")
-- Found BISON: C:/msys64/usr/bin/bison.exe (found version "3.8.2")
-- Looking for libelf.h
-- Looking for libelf.h - found
-- Looking for libelf/libelf.h
-- Looking for libelf/libelf.h - found
-- Looking for usb.h
-- Looking for usb.h - not found
-- Looking for lusb0_usb.h
-- Looking for lusb0_usb.h - found
-- Looking for libusb.h
-- Looking for libusb.h - not found
-- Looking for libusb-1.0/libusb.h
-- Looking for libusb-1.0/libusb.h - found
-- Looking for hidapi/hidapi.h
-- Looking for hidapi/hidapi.h - found
-- Looking for ftdi_tcioflush
-- Looking for ftdi_tcioflush - found
-- Found SWIG: C:/msys64/mingw64/bin/swig.exe (found suitable version "4.2.1", minimum required is "4.0") found components: python
-- Found Python3: C:/Program Files/WindowsApps/PythonSoftwareFoundation.Python.3.13_3.13.1520.0_x64__qbz5n2kfra8p0/python3.13.exe (found version "3.13.5") found components: Interpreter Development Development.Module Development.Embed
CMake Warning (dev) at src/CMakeLists.txt:424 (install):
  Policy CMP0177 is not set: install() DESTINATION paths are normalized.  Run
  "cmake --help-policy CMP0177" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at src/CMakeLists.txt:425 (install):
  Policy CMP0177 is not set: install() DESTINATION paths are normalized.  Run
  "cmake --help-policy CMP0177" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuration summary:
-- ----------------------
-- DO HAVE    libelf
-- DO HAVE    libusb
-- DO HAVE    libusb_1_0
-- DO HAVE    libhidapi
-- DON'T HAVE libftdi
-- DO HAVE    libftdi1
-- DO HAVE    libreadline
-- DO HAVE    libserialport
-- DISABLED   doc
-- DISABLED   parport
-- DISABLED   linuxgpio
-- DISABLED   linuxspi
-- DO HAVE    swig+Python3
-- ----------------------
-- Configuring done (8.3s)
-- Generating done (0.1s)
-- Build files have been written to: C:/work/avr/avrdude_main/build_mingw64_nt-10.0-26100
[133/172] Building C object src/CMakeFiles/swig_avrdude.dir/CMakeFiles/swig_avrdude.dir/libavrdudePYTHON_wrap.c.obj
C:/work/avr/avrdude_main/build_mingw64_nt-10.0-26100/src/CMakeFiles/swig_avrdude.dir/libavrdudePYTHON_wrap.c: In function 'swig_varlink_type':
C:/work/avr/avrdude_main/build_mingw64_nt-10.0-26100/src/CMakeFiles/swig_avrdude.dir/libavrdudePYTHON_wrap.c:1514:5: warning: missing initializer for field 'tp_versions_used' of 'PyTypeObject' {aka 'const struct _typeobject'} [-Wmissing-field-initializers]
 1514 |     };
      |     ^
In file included from C:/Program Files/WindowsApps/PythonSoftwareFoundation.Python.3.13_3.13.1520.0_x64__qbz5n2kfra8p0/Include/object.h:1227,
                 from C:/Program Files/WindowsApps/PythonSoftwareFoundation.Python.3.13_3.13.1520.0_x64__qbz5n2kfra8p0/Include/Python.h:72,
                 from C:/work/avr/avrdude_main/build_mingw64_nt-10.0-26100/src/CMakeFiles/swig_avrdude.dir/libavrdudePYTHON_wrap.c:198:
C:/Program Files/WindowsApps/PythonSoftwareFoundation.Python.3.13_3.13.1520.0_x64__qbz5n2kfra8p0/Include/cpython/object.h:232:14: note: 'tp_versions_used' declared here
  232 |     uint16_t tp_versions_used;
      |              ^~~~~~~~~~~~~~~~
C:/work/avr/avrdude_main/build_mingw64_nt-10.0-26100/src/CMakeFiles/swig_avrdude.dir/libavrdudePYTHON_wrap.c: In function 'SwigPyObject_TypeOnce':
C:/work/avr/avrdude_main/build_mingw64_nt-10.0-26100/src/CMakeFiles/swig_avrdude.dir/libavrdudePYTHON_wrap.c:2144:5: warning: missing initializer for field 'tp_versions_used' of 'PyTypeObject' {aka 'const struct _typeobject'} [-Wmissing-field-initializers]
 2144 |     };
      |     ^
C:/Program Files/WindowsApps/PythonSoftwareFoundation.Python.3.13_3.13.1520.0_x64__qbz5n2kfra8p0/Include/cpython/object.h:232:14: note: 'tp_versions_used' declared here
  232 |     uint16_t tp_versions_used;
      |              ^~~~~~~~~~~~~~~~
C:/work/avr/avrdude_main/build_mingw64_nt-10.0-26100/src/CMakeFiles/swig_avrdude.dir/libavrdudePYTHON_wrap.c: In function 'SwigPyPacked_TypeOnce':
C:/work/avr/avrdude_main/build_mingw64_nt-10.0-26100/src/CMakeFiles/swig_avrdude.dir/libavrdudePYTHON_wrap.c:2360:5: warning: missing initializer for field 'tp_versions_used' of 'PyTypeObject' {aka 'const struct _typeobject'} [-Wmissing-field-initializers]
 2360 |     };
      |     ^
C:/Program Files/WindowsApps/PythonSoftwareFoundation.Python.3.13_3.13.1520.0_x64__qbz5n2kfra8p0/Include/cpython/object.h:232:14: note: 'tp_versions_used' declared here
  232 |     uint16_t tp_versions_used;
      |              ^~~~~~~~~~~~~~~~
[172/172] Linking C shared module src\_swig_avrdude.pyd

Build succeeded.

Run

sudo cmake --build build_mingw64_nt-10.0-26100 --target install

to install.
```

**Comment by stefanrueger** [comment]
For all I know these warnings could well be serious indicators of the wrong headers being used for some libraries or another. Does the built gui work?

**Reply by mcuee** [reply]
> For all I know these warnings could well be serious indicators of the wrong headers being used for some libraries or another. Does the built gui work?

The GUI works well.

**Edit: need to check again. Last time when I tested it, it worked. But now I need to double check how I get it working again.**

**Reply by mcuee** [reply]
The recommended Python GUI build is with MSVC version though, as the Python.org Python (including Windows Store version) is more compatible with MSVC.

No warnings for avrdude Python GUI build under MSVC (VS2022).

**Reply by mcuee** [reply]
And I will say it is still a **bug in avrdude CMake script** that it detects Windows Python installation and not MSYS2 mingw's own Python installation.

**Reply by mcuee** [reply]
@stefanrueger 

Hmm, indeed I could not get the Python binding working now under MSYS2 mingw64.

```
$ python ./src/python/adgui.py
Traceback (most recent call last):
  File "C:/work/avr/avrdude/src/python/adgui.py", line 51, in <module>
    import swig_avrdude as ad
  File "C:/msys64/mingw64/lib/python3.12/site-packages/swig_avrdude.py", line 32, in <module>
    import _swig_avrdude
ImportError: DLL load failed while importing _swig_avrdude: The specified module could not be found.

$ python ./src/python/swigtest.py
Traceback (most recent call last):
  File "C:/work/avr/avrdude/src/python/swigtest.py", line 72, in <module>
    import swig_avrdude as ad
  File "C:/msys64/mingw64/lib/python3.12/site-packages/swig_avrdude.py", line 32, in <module>
    import _swig_avrdude
ImportError: DLL load failed while importing _swig_avrdude: The specified module could not be found.
```

It does not work under Python.org 3.12 Virtual Env either.
```
(py312venv) PS C:\work\avr\avrdude_bin\github_bin\mingw64> python.exe .\build_mingw64\src\python\adgui.py
Traceback (most recent call last):
  File "C:\work\avr\avrdude_bin\github_bin\mingw64\build_mingw64\src\python\adgui.py", line 51, in <module>
    import swig_avrdude as ad
  File "C:\work\python\py312venv\Lib\site-packages\swig_avrdude.py", line 32, in <module>
    import _swig_avrdude
ImportError: DLL load failed while importing _swig_avrdude: The specified module could not be found.
```

I remember I did get it working last time when testing PR #1714 but looking back I mainly tested under virtual environment with the MSVC x64 build.

I mentioned in two comments that I got the Python binding including GUI working under MSYS2 mingw64 in the comments of PR #1714. But now I see both were actually using the MSVC x64 build.
* https://github.com/avrdudes/avrdude/pull/1714#issuecomment-2029190494 (Python command line).
* https://github.com/avrdudes/avrdude/pull/1714#issuecomment-2029237621 (Python GUI)

**Reply by mcuee** [reply]
@stefanrueger 

Just wondering if you want to disable MSYS2 related Python build for the avrdude 8.1 release. Thanks.

**Reply by mcuee** [reply]
We have disabled SWIG/Python build for Windows MSYS2 build.
https://github.com/avrdudes/avrdude/blob/main/.github/workflows/build.yml

**Comment by stefanrueger** [comment]
Happy to take your lead, @mcuee

**Reply by mcuee** [reply]
> Happy to take your lead, @mcuee

Let me debug the issue a bit today. If I cannot get it work, then I will have a PR to disable Python binding build for MSYS2 mingw.

By right it should work, I was doing similar things for libftdi project for quite some years (2011-2020): using MSYS2 mingw to build Python binding for Python.org Python distributions.
https://sourceforge.net/projects/picusb/files/

Interestingly, it seems to me that @dl8dtl was using my packaged version of libftdi 1.3git version (`libftdi1-1.3git_devkit_mingw32_13Dec2015`) in the following avrdude cross-build script.
https://github.com/avrdudes/avrdude/blob/main/tools/build-mingw32.sh

**Reply by mcuee** [reply]
> Happy to take your lead, @mcuee

@stefanrueger 

As of now, two issues are present and I cannot get Python GUI to work under MSYS2 mingw. Therefore I think it is better to disable the build in CI for avrdude 8.1 release.

PR #2036 has been rasied.

1) It does not detect the right MSYS2 mingw Python.
2) Even if I fix the first issue and get the Python binding working, there is a PySide6 packaging issue which prevents the Python GUI to work properly.

Reference:
* https://github.com/avrdudes/avrdude/issues/1946#issuecomment-2994948635

**Comment by stefanrueger** [comment]
This discussion is for known compiler warnings that AVRDUDE cannot do anything about. I have deleted comments about compiler warnings that could and were fixed in AVRDUDE

**Comment by mcuee** [comment]
Looks like no more warnings under macOS with latest git build.
https://github.com/avrdudes/avrdude/actions/runs/33599728349/job/100150531392
